designer/apps/pages/models.py

```diff
@@ -215,6 +223,15 @@ class EnterprisePage(Page):
         InlinePanel('branding', label="Enterprise Page Branding", max_num=1),
     ]
 
+    @classmethod
+    def can_exist_under(cls, parent):
+        can_exist_under_parent = super(EnterprisePage, cls).can_exist_under(parent)
+        parent_child_pages = parent.get_descendants()
```

Comment:
Is this effectively doing the check to see if this page has siblings?

Reply:
Yeah, this determines if a new page has any siblings by grabbing its parent page's existing descendants (i.e, child pages).

Reply:
So my first thought was, this page should be returned in get_descendents() (which would make parent_child_pages truthy). But this is a pre_save hook, so that's why the page doesn't "exist" yet, because it hasn't been saved?

Reply:
This is not so much a pre_save hook insofar as it prevents a user from even viewing the page creation form in the CMS if a page does not pass the `can_exist_under` check on `ProgramPage` and `EnterprisePage`. And, you are correct in your thinking why the page is not returned in `get_descendents`.

Reply:
Gotcha, cool thanks